app.py

```diff
@@ -39,43 +40,60 @@
 
 # Setup Logging
 logging.basicConfig(level=os.environ.get("LOGLEVEL", "INFO"),
-                    format='%(levelname)s: %(message)s')
+                    format='[%(asctime)s] %(levelname)s: %(message)s',
+                    datefmt='%m/%d/%y %H:%M:%S')
 
 # Global variables
 static_img_path = "static/img/images/"
 temp_img_prefix = "MAX-"
 image_captions = collections.OrderedDict()
-VALID_EXT = ['png', 'jpg', 'jpeg']
+VALID_EXT = ['.png', '.jpg', '.jpeg']
 error_raised = []
+app_cookie = 'max-image-caption-generator-web-app'
 
 
-class MainHandler(web.RequestHandler):
+class BaseHandler(web.RequestHandler):
+    def prepare(self):
+        # Documentation on cookies:
+        # http://www.tornadoweb.org/en/stable/guide/security.html
+        if not self.get_cookie(app_cookie):
```

Comment:
You probably want to use `get_secure_cookie()` here to prevent advanced users from viewing each others' images. See http://www.tornadoweb.org/en/stable/guide/security.html

Reply:
I considered that, but if I use the secure cookie then I can't use it's value in the user's uploaded file names since it's no longer a clean string. In which case I would have to keep a data structure containing a list of users and their files, complicating the code quite a bit.

Reply:
Could you create a simple UUID to store in the secure cookie to use as the filename string? Then it would be a one-to-one from the secure cookie to the uuid

Reply:
Based on a better understanding of the library I was able to switch back to secure cookies, just pushed the update for review @MLnick @frreiss I also made the cookie unique to eat instance as well